Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -27,45 +28,82 @@ namespace ROS2
     }
 
-    void RobotImporterWidget::OnModelLoaded()
+    void RobotImporterWidget::ReportError(AZStd::string errorMessage)
     {
-        m_robotNameLabel.setText(m_urdfModel->getName().c_str());
+        QMessageBox::critical(this, QObject::tr("Error"), errorMessage.c_str());
+        m_logLabel.setText(errorMessage.c_str());
+        AZ_Error("RobotImporterWidget", false, errorMessage.c_str());
+    }
+    void RobotImporterWidget::ReportWarning(AZStd::string warningMessage)
+    {
+        AZ_Warning("RobotImporterWidget", false, warningMessage.c_str());
+    }
+    void RobotImporterWidget::ReportInfo(AZStd::string infoMessage)
+    {
+        m_logLabel.setText(infoMessage.c_str());
+        AZ_TracePrintf("RobotImporterWidget", infoMessage.c_str());
+    }
+    AZStd::string RobotImporterWidget::GetURDFPath()
+    {
+        return GetPathWithExtension("Unified Robot Description Format (*.urdf)", QFileDialog::ExistingFiles);
+    }
 
-        URDFPrefabMaker prefabMaker(AZStd::string(m_robotFileNameLabel.text().toUtf8().constData()), m_urdfModel);
-        auto outcome = prefabMaker.CreatePrefabFromURDF();
-        if (!outcome)
-        { // TODO - handle, show
-            AZ_Error("RobotImporterWidget", false, "Importing robot definition failed with error: %s", outcome.GetError().c_str());
+    AZStd::string RobotImporterWidget::GetNewPrefabPath()
+    {
+        return GetPathWithExtension("Prefab (*.prefab)", QFileDialog::AnyFile);
+    }
+
+    RobotImporterInputInterface::ExistingPrefabAction RobotImporterWidget::GetExistingPrefabAction()
+    {
+        QMessageBox msgBox;
+        msgBox.setText("The prefab file already exists");
+        msgBox.setInformativeText("Do you want to overwrite it or choose a new filename?");
+        msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Discard | QMessageBox::Cancel);
+        msgBox.setDefaultButton(QMessageBox::Save);
+        msgBox.setButtonText(QMessageBox::Save, "Overwrite");
+        msgBox.setButtonText(QMessageBox::Discard, "Save As..");
+        auto ret = msgBox.exec();
+        switch (ret)
+        {
+        case QMessageBox::Save:
+            return ExistingPrefabAction::Overwrite;
+        case QMessageBox::Discard:
+            return ExistingPrefabAction::NewName;
+        case QMessageBox::Cancel:
+            return ExistingPrefabAction::Cancel;
+        default:
+            return ExistingPrefabAction::Cancel;
+        }
+    }
+
+    AZStd::string RobotImporterWidget::GetPathWithExtension(AZStd::string extensionDescription, QFileDialog::FileMode mode)
+    {
+        QFileDialog m_importFileDialog;
+        m_importFileDialog.setFileMode(mode);
+        m_importFileDialog.setNameFilter(QObject::tr(extensionDescription.c_str()));
+        m_importFileDialog.setViewMode(QFileDialog::Detail);
+
+        int result = m_importFileDialog.exec();
+        if (result != QDialog::DialogCode::Accepted)
+        {
+            return {};
```

Comment:
Use AZStd::optional if failure is an option

Reply:
Applied this. Now the nullopt is returned when the user cancels. Also, this change led to a bit more changes, for example, I moved the path existence validation to the GUI side. IMO, it would be advisable to take a look at the entire code, again since it was moved around a bit.